src/compile/mark/common.ts

```diff
@@ -21,8 +21,8 @@ export function applyColorAndOpacity(e: VgEncodeEntry, model: UnitModel) {
     applyMarkConfig(e, model, STROKE_CONFIG);
   }
 
-  let colorRef= ref.midPoint('color', model.encoding().color, model.scaleName('color'), model.scale('color'), undefined);
-  let opacityRef = ref.midPoint('opacity', model.encoding().opacity, model.scaleName('opacity'), model.scale('opacity'), {value: config.mark.opacity});
+  let colorRef = ref.midPoint('color', model.encoding().color, model.scaleName('color'), model.scale('color'), undefined);
+  let opacityRef = ref.midPoint('opacity', model.encoding().opacity, model.scaleName('opacity'), model.scale('opacity'), config.mark.opacity && {value: config.mark.opacity});
```

Comment:
I wonder why using 
```
config.mark.opacity && {value: config.mark.opacity}
```
doesn't fail the TS compiler since `config.mark.opacity && {value: config.mark.opacity}` can be `false` and does not match the type...